### PR TITLE
#1177 - Revise decompose interface

### LINKED ISSUE #1177: Revise decompose interface
If the options provided to [decompose](https://github.com/JuliaReach/LazySets.jl/blob/master/src/Approximations/decompositions.jl#L219) are incompatible, it chooses to ignore some and proceed. But this choice is to a certain extent arbitrary. 

Proposal: return an error if the options passed to `decompose` are incompatible, see the [Notes](https://github.com/JuliaReach/LazySets.jl/blob/master/src/Approximations/decompositions.jl#L69) for examples.

### PR DESCRIPTION
**Pull request by schillic**
Closes #1177.

* [x] new interface
* [x] add convenience methods
* [x] adapt tests
* [x] add parts of the old docs with the (adapted) examples again
* [x] add option `block_size`

### CODE REVIEW

File: src/Approximations/decompositions.jl [hunk 1]
```diff
@@ -1,60 +1,37 @@
...
-- `blocks`      -- (optional, default: [2, …, 2] or [1, …, 1] if `set_type` is an interval)
-                   block structure - a vector with the size of each block
-- `block_types` -- (optional, default: Interval for 1D and Hyperrectangle
-                   for mD blocks) a mapping from set types to blocks
-- `directions`  -- (optional, default: `nothing`) template direction type, or
-                   `nothing`
+- `S`         -- set
+- `partition` -- vector of blocks (i.e., of vectors of integers) (see the Notes
+                 below)
+- `block2oa`  -- mapping from block indices in `partition` to a corresponding
```
Comment (mforets) [inline src/Approximations/decompositions.jl]: can we use a more descriptive name for `block2oa`? like `options`, `block_options`, `set_types`, `block_types`,..

File: src/Approximations/decompositions.jl [hunk 2]
```diff
@@ -63,89 +40,78 @@ projections.
...
 
-julia> S = Ball2(zeros(4), 1.);
+julia> S = Ball2(zeros(4), 1.);  # set to be decomposed (4D unit ball in the 2-norm)
 
-julia> array(decompose(S))
-2-element Array{LazySet{Float64},1}:
- Hyperrectangle{Float64}([0.0, 0.0], [1.0, 1.0])
- Hyperrectangle{Float64}([0.0, 0.0], [1.0, 1.0])
-```
+julia> P2d = [1:2, [3, 4]];  # a partition with two blocks of size two
```
Comment (mforets) [inline src/Approximations/decompositions.jl]: why not either `[[1,2], [3, 4]]` or `[1:2, 3:4]` ?

**Comment by mforets** [comment]
Looks good!

File: src/Approximations/decompositions.jl (hunk 2 above)
Comment (schillic) [reply src/Approximations/decompositions.jl]: Ha, this was on purpose to show the most popular ways, but I see now that is confusing.

**Pull request merged**